web/src/main/java/org/apache/shiro/web/filter/authc/FormAuthenticationFilter.java

```diff
@@ -30,6 +31,39 @@
 import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
 
+/**
+
+ * The LoginUtils class provides utility methods for handling login functionality in a web application.
+ * It contains a method for detecting whether a login submission has been made via an HTTP POST request,
+ * which can be overridden by subclasses to provide custom behavior for detecting login submissions.
+ * <p>
+ * Subclasses can modify this behavior to detect login submissions via an HTTP PUT request or other HTTP methods.
+ * </p>
+ * <p>
+ * This class is designed to work together with the AccessControlFilter, which enforces access control policies in a web application.
+ * The AccessControlFilter can use the isLoginSubmission() method to determine whether a request is a login submission
+ * and should be handled differently from other requests.
+ * </p>
+ * <p>
+ * This class is not intended to be instantiated or subclassed, as all of its methods are static and protected.
+ * </p>
+ * @since 0.9
+ */
+class LoginUtils {
+    /**
```

Comment:
Why was this introduced? It's not really deserving of a complete class

Reply:
By moving the isLoginSubmission method to a separate class, we have made the code more modular and improved the cohesion of the classes. Additionally, any other classes that need to check if a request is a login submission can now use the LoginUtils class, rather than duplicating the logic in their own code. I think this could make the code more readable.

Reply:
Sorry I disagree. Classic case of over engineering

Reply:
Please revert this refactor